src/withLocation.tsx

```diff
@@ -0,0 +1,12 @@
+import React from 'react'
+import { useLocation } from 'react-router-dom'
+
+export type WithLocationPropType = {
+  location: Location
+}
+
+export default function withLocation(
+  Component: React.ComponentType<WithLocationPropType>,
+) {
+  return (props) => <Component {...props} location={useLocation()} />
+}
```

Comment:
Just passes location data from the hook. Allows us to replace `withRouter` for `AppInitializer`